msprime/likelihood.py

```diff
@@ -28,9 +28,15 @@ def log_mutation_likelihood(ts, mutation_rate):
     """
     Returns the unnormalised log probability of the stored pattern of mutations
     on the stored tree sequence, assuming infinite sites mutation. In particular,
-    each stored site must only contain a single mutation. The omitted normalising
-    constant depends on the pattern of mutations, but not on the tree sequence or
-    the mutation rate.
+    each stored site must only contain a single mutation, and all mutant alleles
+    are treated interchangeably as having arisen from a single mutation rate.
+    The omitted normalising constant depends on the pattern of mutations, but not
+    on the tree sequence or the mutation rate.
+
+    .. warning::
+        This method only supports continuous genomes.
```

Comment:
How about "The infinite sites assumption means that this method is only valid for continuous genomes"?

We could also say "(and an error will be output if it is attempted on discrete genomes)", if that's true (I don't know if it is!)

Reply:
I like the rewording.

An error hasn't been implemented, but it would be a good idea to add it to both the likelihood methods if it is a simple thing to do. I guess we could either use MSP_ERR_BAD_PARAM_VALUE, or define a whole new error.

Reply:
Thanks. I suspect it's difficult to detect if you are in a (mutationally) continuous genome or not, other than by checking if all the site positions are integers.

I guess people might want to calculate the likelihood even on a discrete genome, if they think the mutation rate is very low, so maybe it should be a warning, not an error?

Reply:
Are there any warnings (other than MSP_WARN_UNUSED) in the source code as it is? I have no idea how to generate one correctly, and could use an example.

Reply:
Let's not bother with this - a warning to the user and an explanation what will happen if you run it on discrete coordinates will do for now.

Reply:
As in, in the documentation.